src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs

```diff
@@ -299,12 +326,26 @@ private async Task PreviewAndExecuteUpdateActionsForSinglePackage()
                     new GatherCache(),
                     sourceCacheContext);
 
+                // PackageReference projects don't support `Update-Package -Reinstall`. 
+                var relevantProjects = Projects;
+                if (Reinstall.IsPresent)
```

Comment:
This one looks exactly same as above could we make separate method and just call from both places.

Reply:
It's very similar, I'll refactor once the concept has been approved.